subsys/nfc/lib/platform.c

```diff
@@ -51,6 +70,95 @@ nrfx_err_t nfc_platform_setup(void)
 }
 
 
+static nrfx_err_t nfc_platform_tagheader_get(uint8_t index,
+					     uint32_t *tag_header)
+{
+	const uint32_t tag_header_addresses[] = {
+		(uint32_t) &NRF_FICR->NFC.TAGHEADER0,
```

Comment:
@kapi-no is FICR accessed by non-secure code here?

Reply:
Or it is just to extract the address ranges?

Reply:
No, its not. I am getting FICR location at this point which I later use for secure service API call at L90

Reply:
I think it would be cleaner to have here:
```
#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER0,
		...
#else
		(uint32_t) &NRF_FICR->NFC.TAGHEADER0,
		...
```
instead of defining `NRF_FICR` to `NRF_FICR_S` in [line 33](https://github.com/nrfconnect/sdk-nrf/pull/2368/files#diff-7d83d6f9a4f1c7ed015c2e85a7f5da12R33).

BTW, `CONFIG_TRUSTED_EXECUTION_NONSECURE` may be undefined, so it should be rather checked with `#ifdef` or `#if IS_ENABLED()`. Some linters may be unhappy otherwise.

Reply:
I improved it according to your suggestion, thanks